src/client.ts

```diff
@@ -455,7 +455,7 @@ interface IUploadKeySignaturesResponse {
 }
 
 export interface IPreviewUrlResponse {
-    [key: string]: string | number;
+    [key: string]: string | number | undefined;
```

Comment:
this seems a bit silly to type - what's the error from typescript?

Reply:
6 of the values are marked as optional with `?` (such as `"og:image"?`) so tsc says:
```
node_modules/matrix-js-sdk/lib/client.d.ts:305:5 - error TS2411: Property '"og:image"' of type 'string | undefined' is not assignable to 'string' index type 'string | number'.
```